web-ui/src/backup_account/backup_email/backup_email.js

```diff
@@ -54,8 +54,14 @@ export class BackupEmail extends React.Component {
       body: JSON.stringify({
         csrftoken: [browser.getCookie('XSRF-TOKEN')]
       })
-    }).then(() => this.props.onSubmit('success'));
-  }
+    }).then((response) => {
+      if (response.status === 204) {
+        this.props.onSubmit('success');
+      } else {
+        this.props.onSubmit('error');
+      }
+    });
```

Comment:
They do not have the `catch` based on the HTTP response.  To use the `catch`,  they've used (in their examples) a separate callback to raise an 'Error' based on the HTTP status, which seems a little long-winded way of doing the same thing.